src/sentry/search/snuba/executors.py

```diff
@@ -564,3 +569,244 @@ def calculate_hits(
+        "date": "events.last_seen",
+        "freq": "times_seen",
+        "new": "events.first_seen",
+        "priority": "priority",
+    }
+
+    aggregation_defs = {
+        "times_seen": ["count()", ""],
+        "events.first_seen": ["multiply(toUInt64(min(events.timestamp)), 1000)", ""],
+        "events.last_seen": ["multiply(toUInt64(max(events.timestamp)), 1000)", ""],
+        "priority": ["toUInt64(plus(multiply(log(times_seen), 600), `events.last_seen`))", ""],
+        "total": ["uniq", ISSUE_FIELD_NAME],
+    }
+
+    @property
+    @abstractmethod
+    def dataset(self):
+        return snuba.Dataset.Groups
+
+    def _transform_converted_filter(
+        self, search_filter, converted_filter, project_ids=None, environment_ids=None
+    ):
+
+        special_date_names = ["active_at", "first_seen", "last_seen"]
+        if search_filter.key.name in special_date_names:
+            # Need to get '2018-02-06T03:35:54' format out of 1517888878000 format
+            datetime_value = datetime.fromtimestamp(converted_filter[2] / 1000)
+            datetime_value = datetime_value.replace(microsecond=0).isoformat().replace("+00:00", "")
+            converted_filter[2] = datetime_value
+
+        # TODO: There is a better way to do this...the issue is that the table/alias is forked on environments, which can't be used in constrain_column_to_dataset
+        if search_filter.key.name in ["first_seen", "last_seen"]:
+            if environment_ids is not None:
+                # Remove as a snuba filter, since we are going to be applying this filter with pre/post filter on Postgres
+                # Since we have GroupEnvironment data that persists beyond retention, whereas Snuba only keeps events in the retention period.
+                return None, None
+            else:
+                table_alias = "groups."
+
+            if isinstance(converted_filter[0], list):
+                converted_filter[0][1][0] = table_alias + converted_filter[0][1][0]
+            else:
+                converted_filter[0] = table_alias + converted_filter[0]
+
+        # if search_filter.key.name == "first_release":
+        #     # The filter's value will be the release's "version". Snuba only knows about ID. So we convert version to id here.
+        #     release = Release.objects.filter(
+        #         version=converted_filter[2],
+        #         organization_id=Project.objects.get(id=project_ids[0]).organization_id,
+        #     )
+        #     if not release:
+        #         # TODO: This means there will be no results and we do not need to run this query!
+        #         # Temp solution is to set value to -1, which should never be a real release ID.
+        #         converted_filter[2] = -1
+        #     else:
+        #         converted_filter[2] = release[0].id
+
+        return converted_filter
+
+    def query(
```

Comment:
There seems to be a lot of duplication between this method and the equivalent in PostgresSnubaQueryExecutor. Could yo uplease factor the common logic into the parent class so that it would be harder to introduce bugs on one side.